test/jdk/sun/tools/jcmd/TestJcmdPIDSubstitution.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, Red Hat Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8334492
+ * @summary Test to verify jcmd accepts %p in output filenames and substitutes for PID
+ * @library /test/lib
+ * @modules java.base/jdk.internal.misc
+ *          java.management
+ * @run main/othervm TestJcmdPIDSubstitution
+ *
+ */
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class TestJcmdPIDSubstitution {
+
+    private static final String FILENAME = "myfile%p";
+
+    public static void main(String[] args) throws Exception {
+        verifyOutputFilenames("Thread.dump_to_file", FILENAME);
+        verifyOutputFilenames("GC.heap_dump", FILENAME);
+        verifyOutputFilenames("Compiler.perfmap", FILENAME);
+        verifyOutputFilenames("System.dump_map", "-F=%s".formatted(FILENAME));
+    }
+
+    private static void verifyOutputFilenames(String... args) throws Exception {
+        long pid = ProcessTools.getProcessId();
+        Path path;
+        do {
+            path = Paths.get("myfile%d".formatted(pid));
+        } while(Files.exists(path));
```

Comment:
Why this do/while loop is needed? Each test should have it's own empty scratch directory.